### ISSUE TITLE
splatted arguments and keyword argument in block causes missing keyword error

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1613]
I'm not sure how to describe it by words but the code is clear. This construct is heavily used in dry and ROM.

To reproduce:

```ruby
def demo(*args)
  yield(*args, type_spec: true)
end

demo do |*args, type_spec:|
  puts *args
  puts type_spec
end
```

Error:

```
block.rb:5:in `block in <main>': missing keyword: type_spec (ArgumentError)
	from block.rb:2:in `demo'
	from block.rb:5:in `<main>'
```

**Comment by deepj** [comment]
@chrisseaton please, make this as a high priority for RC14. This solves many issues in dry and ROM land

**Comment by chrisseaton** [comment]
I don't think our yield handles keyword arguments properly. It's kind of a parallel to the usual method call, but with all the same complexity and less-well tested.

**Comment by deepj** [comment]
@chrisseaton: can be this fixed in RC15? I've added testing Truffleruby in Travis CI for other several dry-rb gems and this would fixed a lot of the rest issues.

**Comment by deepj** [comment]
Is there any change to be this fixed in RC15?

**Comment by deepj** [comment]
@eregon Hello, is there any change to get fixed this into RC15?

**Comment by chrisseaton** [comment]
Sorry this hasn't managed to go into RC15, but we have a lot of time scheduled this month to look at this issue backlog.

**Comment by deepj** [comment]
😿

**Comment by chrisseaton** [comment]
I should be able to get you a build with it in a week or so though.

**Comment by chrisseaton** [comment]
I'm working with this spec for this at the moment

```ruby
  describe "taking |*a, b:|" do
    it "assigns the keyword argument and nothing in rest" do
      @y.k { |*a, b:| [a, b] }.should == [[], true]
    end
  end
```

**Comment by chrisseaton** [comment]
This is probably more precisely the problem

```ruby
  describe "taking a splat and a keyword argument" do
    it "passes it as an array of a hash" do
      @y.k([1, 2]) { |*a| a }.should == [1, 2, {:b=>true}] # [[1, 2, {:b=>true}]] on TruffleRuby
    end
  end
```

**Comment by chrisseaton** [comment]
A fix for this is on the way.

**Comment by deepj** [comment]
@chrisseaton If possible to make a new nightly build for me after fixing this, please?

**Comment by chrisseaton** [comment]
Fixed in 0c71a5bf5d2681cf2acbb875fd779c505f6f55b8.
